bob-backend/src/pearl/group.rs

```diff
@@ -307,13 +319,30 @@ impl Group {
     pub async fn exist(&self, keys: &[BobKey]) -> Vec<bool> {
         let mut exist = vec![false; keys.len()];
         let holders = self.holders.read().await;
+        let mut max_timestamp = None;
+        let mut result = false;
         for (ind, &key) in keys.iter().enumerate() {
             for (_, Leaf { data: holder, .. }) in holders.iter_possible_childs_rev(&Key::from(key))
             {
-                if !exist[ind] {
-                    exist[ind] = holder.exist(key).await.unwrap_or(false);
+                match holder.exist(key).await.unwrap_or(ReadResult::NotFound) {
+                    ReadResult::Found(ts) => {
```

Comment:
Same problem: Pearl level timestamp is used here. We should use user level timestamp instead